File: core/java/src/main/java/hara/lang/protocol/IInvokeIn.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.iinvokein", name = "IInvokeIn")
public interface IInvokeIn {
  @HaraMethod(value = "invoke-in", arity = -1, variadic = true)
  Object invokeIn(IContext context, Object... args);
}

File: core/rust/src/lang/protocol/iinvokein.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.iinvokein", name = "IInvokeIn")]
pub trait IInvokeIn<C, A> {
    type Output;

    #[hara_method(value = "invoke-in", arity = -1, variadic = true, min_arity = 2)]
    fn invoke_in(&self, context: &mut C, arguments: A) -> Self::Output;
}
